hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeDecommissionManager.java

```diff
@@ -391,17 +391,21 @@ private synchronized boolean checkIfDecommissionPossible(List<DatanodeDetails> d
     int numDecom = dns.size();
     List<DatanodeDetails> validDns = new ArrayList<>(dns);
     int inServiceTotal = nodeManager.getNodeCount(NodeStatus.inServiceHealthy());
+    int unHealthyTotal = nodeManager.getNodeCount(NodeStatus.inServiceStale()) +
+        nodeManager.getNodeCount(NodeStatus.inServiceDead());
```

Comment:
Let's move this computation to the `if` block where the error message is being created. So we perform this computation only if needed.

Reply:
I think what we really need is the total of all nodes that are not (in-service and healthy). That is, `totalNodes - inServiceTotal`. This is because nodes that are entering maintenance, in maintenance, decommissioning etc. also don't count toward in-service healthy nodes.

The easiest way I can see to get `totalNodes` is to use the following API
```
List<DatanodeDetails> getAllNodes();
```
and then get the size.

`NodeStateManager` has a `getTotalNodeCount` API but unfortunately it's not exposed in the interface. We can use `getAllNodes` for now. And later, in another jira, we can expose `getTotalNodeCount` in the interface and use it.

Reply:
Fixed based on the comments. Now we will show only no of HEALTHY IN-SERVICE and UNHEALTHY (total_nodes-in_service_healthy) nodes.